### PR TITLE
fix all the remaining eslint warnings

### PR DESCRIPTION
**Pull request by dario-piotrowicz**
continuation from #544 🙂 

Some changes might be controversial I am happy to adjust them any way you prefer, I just applied what I thought more sensible and/or my personal preference 🙂

### CODE REVIEW

File: .eslintrc.cjs [hunk 1]
```diff
@@ -29,8 +29,6 @@ module.exports = {
     ],
 
     "@typescript-eslint/unbound-method": "error",
-
-    "@typescript-eslint/no-non-null-assertion": "warn",
```
Comment (dario-piotrowicz) [inline .eslintrc.cjs:33]: I am personally not a fan of this one 😅 

if someone uses a non-null assertion I think that's generally for a good reason so having to add an eslint disabling comment for it feels very unnecessary/annoying to me 🤷

File: .eslintrc.cjs [hunk 2]
```diff
@@ -15,7 +15,7 @@ module.exports = {
     ],
     "sonarjs/elseif-without-else": "warn",
     "sonarjs/no-duplicate-string": "warn",
-    "sonarjs/cognitive-complexity": "warn",
+    "sonarjs/cognitive-complexity": ["warn", 35],
```
Comment (dario-piotrowicz) [inline .eslintrc.cjs:18]: `35` allows most functions in the codebase, I could just add disabling comment to each one instead if preferred (there were around 10-ish or something I think)

I did not attempt on splitting the functions as I am not really familiar with the codebase here (but I am happy to give it a go if you want 🙂)

File: .eslintrc.cjs (hunk 1 above)
Comment (conico974) [reply .eslintrc.cjs:33]: Yeah I'm fine with this
We just need to verify on every PR to make sure it's not used unwisely

File: .eslintrc.cjs (hunk 2 above)
Comment (conico974) [reply .eslintrc.cjs:18]: I agree.
I need to take a closer look at the PR later today, I remember that there was a reason that it was this order. Might be old code

**Review by conico974 — APPROVED** [review]
Sorry it was old code, there was a reason for the order back when we used multiple file for cache, but now that we have a single file it doesn't matter

**Comment by conico974** [comment]
https://github.com/opennextjs/opennextjs-aws/actions/runs/11312902537/job/31460954988

File: packages/open-next/src/build/generateOutput.ts [hunk 3]
```diff
@@ -92,6 +92,8 @@ interface OpenNextOutput {
   };
 }
 
+const indexHandler = "index.hander";
```
Comment (conico974) [inline packages/open-next/src/build/generateOutput.ts:95]: There was a typo here

Comment (dario-piotrowicz) [reply packages/open-next/src/build/generateOutput.ts:95]: sorry 😓

Comment (conico974) [reply packages/open-next/src/build/generateOutput.ts:95]: No problem

Comment (sommeeeer) [reply packages/open-next/src/build/generateOutput.ts:95]: > sorry 😓

Typos happens to everyone. Your PR made the code more DRY and easier to read. Great work ⭐️

**Pull request merged**